runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java

```diff
@@ -415,17 +415,17 @@ private void lookup()
 					label.setText(Integer.toString(combatLevel));
 				}
 			}
-			else if (result.getSkill(skill) != null && result.getSkill(skill).getRank() != -1)
+			else if (result.getSkill(skill) != null)
 			{
 				Skill s = result.getSkill(skill);
 				int level;
 				if (config.virtualLevels() && SKILLS.contains(skill))
 				{
-					level = Experience.getLevelForXp((int) s.getExperience());
+					level = Experience.getLevelForXp((int) Math.max(0, s.getExperience()));
```

Comment:
Wouldn't this incorrectly show level 1 for people who are unranked/have -1 xp?

Reply:
There is a bug in the RS hiscore API that sometimes returns the data as [-1,1,-1] instead of the true values. This indicates that the skill is unranked, level 1 and with -1 exp. The Math.max() catches this scenario and enforces that the player exp is always within the valid range 0 to 200,000,000.